clickhouse-http-client/src/main/java/com/clickhouse/client/http/HttpUrlConnectionImpl.java

```diff
@@ -146,12 +149,21 @@ private String getResponseHeader(String header, String defaultValue) {
         return value != null ? value : defaultValue;
     }
 
+    private static final Set<String> maskedHeaders = new HashSet<>(Arrays.asList("authorization",
+            "proxy-authorization","x-clickhouse-key"));
+
     private void setHeaders(HttpURLConnection conn, Map<String, String> headers) {
         headers = mergeHeaders(headers);
 
         if (headers != null && !headers.isEmpty()) {
             for (Entry<String, String> header : headers.entrySet()) {
-                log.debug("Adding header key [%s] value [%s]", header.getKey(), header.getValue());
+                if (log.isDebugEnabled()) {
```

Comment:
It is not such a big issue, but headers are a map, and we are converting them to set so we iterate on all the headers that we have (size N) instead of iterating on maskHeders set (size M=2) and only to check if we have the header O(1) get operation of a map maskedHeaders so I think in this case the search should be in size of the maskHeders list.